park-dashboard/src/Components/WeatherForecast/WeatherForecast.js

```diff
@@ -0,0 +1,311 @@
+import React, { useState, useEffect, useRef } from "react";
+import moment from "moment";
+import "./WeatherForecast.css";
+import { WiCloudy, WiRain, WiSnow } from "react-icons/wi"; // Import weather icons from react-icons library
+import {
+  WiDaySunny,
+  WiDayCloudy,
+  WiCloud,
+  WiThunderstorm,
+  WiFog,
+  WiSmoke,
+  WiDayHaze,
+  WiDust,
+  WiSandstorm,
+  WiVolcano,
+  WiStrongWind,
+  WiTornado,
+  WiShowers,
+} from "react-icons/wi";
+import { IoMdRainy } from "react-icons/io";
+import { BsFillCloudSlashFill, BsCloudsFill } from "react-icons/bs";
+import { GiRaining } from "react-icons/gi";
+import { Line } from "react-chartjs-2";
+import ChartDataLabels from "chartjs-plugin-datalabels";
+import Chart from "chart.js/auto";
+
+function WeatherForecast(item) {
+  const [dailyForecast, setDailyForecast] = useState([]);
+  const [latitude, setLatitude] = useState(null);
+  const [longitude, setLongitude] = useState(null);
+  const weatherForecastRef = useRef(null);
+  const parkCode = item.parkCode;
+  useEffect(() => {
+    const fetchWeatherData = async () => {
+      try {
+        const API_KEY = "92dbeb6cb1924c3a12889184257942ea";
+        const UNITS = "metric";
+        const response = await fetch(
+          `https://api.openweathermap.org/data/2.5/forecast?lat=${latitude}&lon=${longitude}&units=${UNITS}&appid=${API_KEY}`
+        );
+        const data = await response.json();
+        const dataList = data.list;
+        const groupedData = groupDataByDay(dataList);
+        const dailyData = calculateDailyData(groupedData);
+        setDailyForecast(dailyData);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    if (latitude && longitude) {
+      fetchWeatherData();
+    }
+  }, [latitude, longitude]);
+
+  useEffect(() => {
+    const fetchParkData = async () => {
+      try {
+        const response = await fetch(
+          `https://developer.nps.gov/api/v1/parks?parkCode=${parkCode}&api_key=OaR7jmqSa22JAcsym9lVfStp58LmCqH9JdZUPEH7`
+        );
+        const data = await response.json();
+        const parkData = data.data;
+
+        if (parkData.length > 0) {
+          setLatitude(parkData[0].latitude);
+          setLongitude(parkData[0].longitude);
+        }
+      } catch (error) {
+        console.error("Error fetching park data:", error);
+      }
+    };
+    fetchParkData();
+  }, []);
+  useEffect(() => {
+    if (weatherForecastRef.current) {
+      const containerWidth = weatherForecastRef.current.offsetWidth;
+      const totalWidth = dailyForecast.length * 200;
+      if (totalWidth > containerWidth) {
+        weatherForecastRef.current.classList.add("scrollable");
+      } else {
+        weatherForecastRef.current.classList.remove("scrollable");
+      }
+    }
+  }, [dailyForecast]);
+
+  const groupDataByDay = (data) => {
+    const groupedData = {};
+    data.forEach((item) => {
```

Comment:
If you use the reduce function instead of foreach, the mutation of groupedData can be avoided. 
And I believe that will also make the code more readable.